plugin/trino-opensearch/src/test/java/io/trino/plugin/opensearch/TestOpenSearchComplexTypePredicatePushDown.java

```diff
@@ -0,0 +1,446 @@
+import org.junit.jupiter.api.Test;
+import org.opensearch.client.Request;
+import org.opensearch.client.RestClient;
+import org.opensearch.client.RestHighLevelClient;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+
+import static io.trino.plugin.opensearch.OpenSearchServer.OPENSEARCH_IMAGE;
+import static io.trino.testing.TestingNames.randomNameSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestOpenSearchComplexTypePredicatePushDown
+        extends AbstractTestQueryFramework
+{
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapperProvider().get();
+
+    private OpenSearchServer opensearch;
+    private RestHighLevelClient client;
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        opensearch = new OpenSearchServer(OPENSEARCH_IMAGE, false, ImmutableMap.of());
+        HostAndPort address = opensearch.getAddress();
+        client = new RestHighLevelClient(RestClient.builder(new HttpHost(address.getHost(), address.getPort())));
+        return OpenSearchQueryRunner.builder(opensearch.getAddress()).build();
+    }
+
+    @AfterAll
+    public final void destroy()
+            throws IOException
+    {
+        opensearch.stop();
+        opensearch = null;
+        client.close();
+        client = null;
+    }
+
+    @Test
+    public void testRowTypeOnlyNullsRowGroupPruning()
+    {
+        String tableName = "test_primitive_column_nulls_pruning_" + randomNameSuffix();
+        @Language("JSON")
+        String properties =
+                """
+                {
+                    "properties": {
+                        "col": {
+                            "type": "long"
+                        }
+                    }
+                }
+                """;
+        StringBuilder payload = new StringBuilder();
+        for (int i = 0; i < 4096; i++) {
+            Map<String, Object> document = new HashMap<>();
+            document.put("col", null);
+            Map<String, Object> indexPayload = ImmutableMap.of("index", ImmutableMap.of("_index", tableName, "_id", String.valueOf(System.nanoTime())));
+
+            String jsonDocument;
+            String jsonIndex;
+            try {
+                jsonDocument = OBJECT_MAPPER.writeValueAsString(document);
+                jsonIndex = OBJECT_MAPPER.writeValueAsString(indexPayload);
+            }
+            catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+            payload.append(jsonIndex).append("\n").append(jsonDocument).append("\n");
+        }
+        try {
+            createIndex(tableName, properties);
+            bulkIndex(tableName, payload.toString());
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
```

Comment:
We can add `throws Exception` to the test method instead. Same for the following catch blocks.